Source/WebGPU/WebGPU/Queue.h

```diff
@@ -96,6 +98,48 @@ class Queue : public WGPUQueueImpl, public ThreadSafeRefCounted<Queue> {
     uint64_t m_completedCommandBufferCount { 0 };
     using OnSubmittedWorkDoneCallbacks = Vector<WTF::Function<void(WGPUQueueWorkDoneStatus)>>;
     HashMap<uint64_t, OnSubmittedWorkDoneCallbacks> m_onSubmittedWorkDoneCallbacks;
+
+    class UnscheduledCommandCount : public ThreadSafeRefCounted<UnscheduledCommandCount> {
+    public:
+        static Ref<UnscheduledCommandCount> create()
+        {
+            return adoptRef(*new UnscheduledCommandCount);
+        }
+
+        void increment()
+        {
+            [m_condition lock];
+            ++m_count;
+            [m_condition unlock];
+        }
+
+        void decrement()
+        {
+            [m_condition lock];
+            ASSERT(m_count > 0);
+            --m_count;
+            [m_condition signal];
+            [m_condition unlock];
+        }
+
+        void waitForCountToReach0()
+        {
+            [m_condition lock];
+            while (m_count)
+                [m_condition wait];
+            [m_condition unlock];
+        }
+
+    private:
+        UnscheduledCommandCount()
+            : m_condition([NSCondition new])
+        {
+        }
+
+        NSCondition *m_condition { nil };
```

Comment:
What is the resolution of NSCondition and can this be implemented without a lock?

Reply:
What does "resolution" mean here?

Reply:
Oh sorry I meant how quickly does `wait` return when `signal` is called from the other thread? Maybe that is an implementation detail outside the scope of this review.